content/tutorials/use-tokens/trade-in-the-decentralized-exchange.md

```diff
@@ -0,0 +1,220 @@
+
+## Prerequisites
+
+<!-- Source for this specific tutorial's interactive bits: -->
+<script src='https://cdn.jsdelivr.net/npm/bignumber.js@9.0.2/bignumber.min.js'></script>
+<script type="application/javascript" src="assets/js/tutorials/trade-in-the-dex.js"></script>
+
+This page provides JavaScript examples that use the [xrpl.js](https://js.xrpl.org/) library. See [Get Started Using JavaScript](get-started-using-javascript.html) for setup instructions.
+
+Since JavaScript works in the web browser, you can read along and use the interactive steps without any setup.
+
+
+## Steps
+{% set n = cycler(* range(1,99)) %}
+
+This tutorial demonstrates how to buy a fungible token in the decentralized exchange by selling XRP. (Other types of trades are possible, but selling a token, for example, requires you to have it first.) The example token used in this tutorial is as follows:
+
+| Currency Code | Issuer | Notes |
+|---|---|---|
+| TST | rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd | A test token pegged to XRP at a rate of approximately 10 XRP per 1 TST. The issuer has existing Offers on the XRP Ledger Testnet to buy and sell these tokens for  |
+
+
+### {{n.next()}}. Connect to Network
+
+You must be connected to the network to submit transactions to it. Additionally, some languages (including JavaScript) require a high-precision number library for performing calculations on currency amounts you may encounter in the ledger. The following code shows how to connect to a public XRP Ledger Testnet server a supported [client library](client-libraries.html) with the appropriate dependencies.
+
+<!-- MULTICODE_BLOCK_START -->
+
+_JavaScript_
+
+{{ include_code("_code-samples/trade-in-the-decentralized-exchange/js/base-with-bignumber.js", language="js") }}
+
+<!-- MULTICODE_BLOCK_END -->
+
+**Note:** The JavaScript code samples in this tutorial use the [`async`/`await` pattern](https://javascript.info/async-await). Since `await` needs to be used from within an `async` function, the remaining code samples are written to continue inside the `main()` function started here. You can also use Promise methods `.then()` and `.catch()` instead of `async`/`await` if you prefer.
+
+For this tutorial, click the following button to connect:
+
+{% include '_snippets/interactive-tutorials/connect-step.md' %}
+
+### {{n.next()}}. Get Credentials
+
+To transact on the XRP Ledger, you need an address and secret key, and some XRP. For development purposes, you can get these on the [{{use_network}}](parallel-networks.html) using the following interface:
+
+{% include '_snippets/interactive-tutorials/generate-step.md' %}
+
+When you're [building production-ready software](production-readiness.html), you should use an existing account, and manage your keys using a [secure signing configuration](set-up-secure-signing.html). The following code shows how to instantiate a class with the wallet instance:
```

Comment:
I'm not quite sure what the last sentence means: "The following code shows how to instantiate a class with the wallet instance".

Maybe better wording would be: "Below shows how you can create a Wallet object to track your account and keys:"

Reply:
Yeah, it was meant to say something like "instantiate an instance of the Wallet class" but something like what you wrote is better.